_posts/2025-02-10-announcing-ballerina-2201.11.0-swan-lake-update-11.md

```diff
@@ -0,0 +1,250 @@
+
+```ballerina
+function fibonacci(int n) returns int {
+   if (n <= 1) {
+       return n;
+   }
+   return fibonacci(n - 1) + fibonacci(n - 2);
+}
+
+
+public function main() {
+   int n = 52; // Adjust this number to increase the load
+   int result = fibonacci(n);
+}
+```
+
+Here is a summary of improvements in CPU-bound operations, GraalVM execution time, file size, and memory usage.
+
+#### CPU Bound Operations
+
+<img alt="Loops U10 Vs U11" src="/images/u11/loops-performance-u10-vs-u11.png">
+<img alt="Loops U10 Vs U11" src="/images/u11/cpu-bound-performance-u10-vs-u11.png">
+
+| Scenario | U10 (Seconds) | U11 (seconds) | Improvement |
+| :---- | :---- | :---- | :---- |
+| Nested Loops with array accessing | 21.87 | 1.04 | **21.03x** |
+| Query Expression with 1M loops | 35.17 | 15.49 | **2.27x** |
+| Fibonacci (n \= 52\) | 202.58 | 108.8 | **1.86x** |
+| Workers | 19.39 | 18.16 | **1.07x** |
+
+
+#### GraalVM Build Time
+
+<img alt="Loops U10 Vs U11" src="/images/u11/graalvm-build-time-u10-vs-u11.png">
+
+
+| U10 (Seconds) | U11 (Seconds) | Improvement |
+| :---- | :---- | :---- |
+| 152.00 | 123.67 | **1.23x** |
+
+#### File Size
+<img alt="Loops U10 Vs U11" src="/images/u11/file-size-u10-vs-u11.png">
+
+|  | U10 (MB) | U11 (MB) | Improvement |
+| :---- | ----- | ----- | :---- |
+| HTTP Module Build File | 3.6 | 2.4 | **33.33%** |
+| Passthrough Executable | 37.7 | 35.4 | **6.10%** |
+
+#### Memory
+
+Performance tests were conducted to evaluate the heap memory usage of Ballerina Update 10 and Update 11. Since garbage collection causes fluctuations in memory usage during execution, this analysis focuses on the peak heap memory recorded for each test, demonstrating a significant reduction in memory consumption.  
+
+<img alt="Loops U10 Vs U11" src="/images/u11/peak-heap-memory-u10-vs-u11.png">
+
+| Scenario | U10 Average(MB) | U11 Average (MB)  | Improvement |
+| :---- | :---- | :---- | :---- |
+| HTTP Service with Workers | 3020.8 | 1945.6 | **35.6% decrease** |
+| HTTP Passthrough | 437 | 359 | **18.0% decrease** |
+| JSON to XML Service | 145.7 | 99.3 | **31.8% decrease** |
+
+### Strand Dump Tool
+
+The strand dump tool has been enhanced to support virtual threads. The report now includes the total strand count, identifies strands running in parallel and sequentially, and provides their corresponding stack traces.
+
+In the updated concurrency model, every strand is directly mapped to a Java virtual thread, and the tool extracts strand-related details from the virtual thread dump. However, since the thread dump does not include virtual thread states, the current version of the strand dump report does not display strand states.
+
+### New Runtime APIs
+
+The Ballerina Runtime Java APIs have been redesigned to enhance seamless integration for Java-to-Ballerina interoperability.   
+Here’s an example of how to invoke a Ballerina function from Java code.
+
+**Ballerina**
```

Comment:
Did we update the java interoperability guide pages with new changes? If so, we better point to that page example sections instead of having two examples here.